pydoctor/test/test_astbuilder.py

```diff
@@ -448,24 +492,81 @@ class C(A):
 
 @systemcls_param
 def test_all_recognition(systemcls: Type[model.System]) -> None:
-    ast = astbuilder._parse(textwrap.dedent('''
+    """The value assigned to __all__ is parsed to Module.all."""
+    mod = fromText('''
     def f():
         pass
     __all__ = ['f']
-    '''))
-    mod = fromAST(ast, systemcls=systemcls)
-    astbuilder.findAll(ast, mod)
+    ''', systemcls=systemcls)
     assert mod.all == ['f']
     assert '__all__' not in mod.contents
 
 @systemcls_param
 def test_all_in_class_non_recognition(systemcls: Type[model.System]) -> None:
-    ast = astbuilder._parse(textwrap.dedent('''
+    """A class variable named __all__ is ignored."""
```

Comment:
The test was introduced at the same time as the `__all__` parsing itself in 611d00a8, so it wasn't in reaction to a reported bug in any case. Maybe the bug was found during development or review, or maybe it never existed but the developer thought it would be an easy mistake to make during later refactoring.

There is no module `__all__` in this test, so I'm not sure "not interfere" is the best description. The potential bug it's testing for is the class `__all__` being mistaken for a module `__all__`. But that means that the desired behavior is that `__all__` inside a class is ignored...

Reply:
Ah OK, in that sense it's not ignored indeed. I'll update the test.